tools/topology/sof-ehl-rt5660.m4

```diff
@@ -0,0 +1,189 @@
+# Schedule 48 frames per 1000us deadline on core 0 with priority 0
+PIPELINE_PCM_ADD(sof/pipe-volume-playback.m4,
+	5, 3, 2, s32le,
+	1000, 0, 0,
+	48000, 48000, 48000)
+
+# Low Latency playback pipeline 4 on PCM 4 using max 2 channels of s32le.
+# Schedule 48 frames per 1000us deadline on core 0 with priority 0
+PIPELINE_PCM_ADD(sof/pipe-volume-playback.m4,
+	6, 4, 2, s32le,
+	1000, 0, 0,
+	48000, 48000, 48000)
+
+# Low Latency playback pipeline 4 on PCM 5 using max 2 channels of s32le.
+# Schedule 48 frames per 1000us deadline on core 0 with priority 0
+PIPELINE_PCM_ADD(sof/pipe-volume-playback.m4,
+	7, 5, 2, s32le,
+	1000, 0, 0,
+	48000, 48000, 48000)
+
+#
+# DAIs configuration
+#
+
+dnl DAI_ADD(pipeline,
+dnl     pipe id, dai type, dai_index, dai_be,
+dnl     buffer, periods, format,
+dnl     frames, deadline, priority, core)
+
+# playback DAI is SSP0 using 2 periods
+# Buffers use s32le format, with 48 frame per 1000us on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-playback.m4,
+	1, SSP, 0, SSP0-Codec,
+	PIPELINE_SOURCE_1, 2, s24le,
+	1000, 0, 0)
+
+# capture DAI is SSP0 using 2 periods
+# Buffers use s32le format, with 48 frame per 1000us on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-capture.m4,
+	2, SSP, 0, SSP0-Codec,
+	PIPELINE_SINK_2, 2, s24le,
+	1000, 0, 0)
+
+# capture DAI is DMIC01 using 2 periods
+# Buffers use s32le format, 1000us deadline on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-capture.m4,
+	3, DMIC, 0, dmic01,
+	PIPELINE_SINK_3, 2, s32le,
+	1000, 0, 0)
+
+# playback DAI is iDisp1 using 2 periods
+# Buffers use s32le format, with 48 frame per 1000us on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-playback.m4,
+	4, HDA, 0, iDisp1,
+	PIPELINE_SOURCE_3, 2, s32le,
+	1000, 0, 0, SCHEDULE_TIME_DOMAIN_TIMER)
+
+# playback DAI is iDisp2 using 2 periods
+# Buffers use s32le format, with 48 frame per 1000us on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-playback.m4,
+	5, HDA, 1, iDisp2,
+	PIPELINE_SOURCE_4, 2, s32le,
+	1000, 0, 0, SCHEDULE_TIME_DOMAIN_TIMER)
+
+# playback DAI is iDisp3 using 2 periods
+# Buffers use s32le format, with 48 frame per 1000us on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-playback.m4,
+	6, HDA, 2, iDisp3,
+	PIPELINE_SOURCE_5, 2, s32le,
+	1000, 0, 0, SCHEDULE_TIME_DOMAIN_TIMER)
+
+# playback DAI is iDisp4 using 2 periods
+# Buffers use s32le format, with 48 frame per 1000us on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-playback.m4,
+	7, HDA, 3, iDisp4,
+	PIPELINE_SOURCE_6, 2, s32le,
+	1000, 0, 0, SCHEDULE_TIME_DOMAIN_TIMER)
+
+# capture DAI is DMIC16k using 2 periods
+# Buffers use s32le format, with 16 frame per 1000us on core 0 with priority 0
+#DAI_ADD(sof/pipe-dai-capture.m4,
+#	8, DMIC, 1, dmic16k,
+#	PIPELINE_SINK_8, 2, s32le,
+#	16, 1000, 0, 0)
+
+# PCM Low Latency, id 0
+dnl PCM_PLAYBACK_ADD(name, pcm_id, playback)
+PCM_DUPLEX_ADD(Headset, 0, PIPELINE_PCM_1, PIPELINE_PCM_2)
+PCM_CAPTURE_ADD(DMIC01, 1, PIPELINE_PCM_3)
```

Comment:
I checked and found a lot of files are using DMIC01. I will rename them all if it is OK to you, such as 
```
./tools/topology/sof-apl-da7219.m4:PCM_CAPTURE_ADD(DMIC01, 99, PIPELINE_PCM_4)
./tools/topology/sof-apl-da7219.m4:VIRTUAL_WIDGET(DMIC01 Rx, out_drv, 3)
./tools/topology/sof-apl-dmic-a2ch-b2ch.m4:# PCM6 <----- DMIC6 (DMIC01)
./tools/topology/sof-apl-dmic-a2ch-b2ch.m4:PCM_CAPTURE_ADD(DMIC01, 6, PIPELINE_PCM_6)
./tools/topology/sof-apl-dmic-a2ch-b4ch.m4:# PCM6 <----- DMIC6 (DMIC01)
./tools/topology/sof-apl-dmic-a2ch-b4ch.m4:PCM_CAPTURE_ADD(DMIC01, 6, PIPELINE_PCM_6)
./tools/topology/sof-apl-dmic-a2ch.m4:# PCM6 <----- DMIC6 (DMIC01)
./tools/topology/sof-apl-dmic-a2ch.m4:PCM_CAPTURE_ADD(DMIC01, 6, PIPELINE_PCM_6)
./tools/topology/sof-apl-dmic-a4ch-b2ch.m4:# PCM6 <----- DMIC6 (DMIC01)
./tools/topology/sof-apl-dmic-a4ch-b2ch.m4:PCM_CAPTURE_ADD(DMIC01, 6, PIPELINE_PCM_6)
./tools/topology/sof-apl-dmic.m4:# PCM6 <----- DMIC6 (DMIC01)
./tools/topology/sof-apl-dmic.m4:PCM_CAPTURE_ADD(DMIC01, 6, PIPELINE_PCM_6)
./tools/topology/sof-apl-eq-dmic.m4:# PCM6 <---- EQ IIR <----- DMIC6 (DMIC01)
./tools/topology/sof-apl-eq-dmic.m4:PCM_CAPTURE_ADD(DMIC01, 6, PIPELINE_PCM_1)
./tools/topology/sof-apl-keyword-detect.m4:# PCM 0 <-------+- KPBM 0 <-- B0 <-- DMIC6 (DMIC01)
./tools/topology/sof-apl-nocodec.m4:# PCM6 <---- volume <----- DMIC6 (DMIC01)
./tools/topology/sof-apl-nocodec.m4:PCM_CAPTURE_ADD(DMIC01, 6, PIPELINE_PCM_13)
./tools/topology/sof-apl-src-dmic.m4:# PCM6 <--- Volume <--- SRC <--- DMIC6 (DMIC01)
./tools/topology/sof-apl-src-dmic.m4:PCM_CAPTURE_ADD(DMIC01, 6, PIPELINE_PCM_6)
./tools/topology/sof-cml-demux-rt5682.m4:# PCM1 ----> volume -----> DMIC01 (dmic0 capture)
./tools/topology/sof-cml-demux-rt5682.m4:# capture DAI is DMIC01 using 2 periods
./tools/topology/sof-cml-demux-rt5682.m4:PCM_CAPTURE_ADD(DMIC01, 1, PIPELINE_PCM_3)
./tools/topology/sof-cml-rt5682-kwd.m4:# PCM1 <------------------ DMIC01 (dmic0 capture, , BE link 1)
./tools/topology/sof-cml-rt5682-kwd.m4:# capture DAI is DMIC01 using 2 periods
./tools/topology/sof-cml-rt5682-kwd.m4:PCM_CAPTURE_ADD(DMIC01, 1, PIPELINE_PCM_3)
./tools/topology/sof-cml-rt5682.m4:# PCM1 <------------------ DMIC01 (dmic0 capture, , BE link 1)
./tools/topology/sof-cml-rt5682.m4:# capture DAI is DMIC01 using 2 periods
./tools/topology/sof-cml-rt5682.m4:PCM_CAPTURE_ADD(DMIC01, 1, PIPELINE_PCM_3)
./tools/topology/sof-cml-rt700.m4:# PCM2 <---------------- DMIC01 (dmic0 capture, BE dailink 2)
./tools/topology/sof-cml-rt700.m4:# capture DAI is DMIC01 using 2 periods
./tools/topology/sof-cml-rt700.m4:PCM_CAPTURE_ADD(DMIC01, 2, PIPELINE_PCM_3)
./tools/topology/sof-cml-src-rt5682.m4:# PCM1 <---- volume <----- DMIC01 (dmic0 capture)
./tools/topology/sof-cml-src-rt5682.m4:# capture DAI is DMIC01 using 2 periods
./tools/topology/sof-cml-src-rt5682.m4:PCM_CAPTURE_ADD(DMIC01, 1, PIPELINE_PCM_3)
./tools/topology/sof-cnl-nocodec.m4:# PCM3 <---- volume <----- DMIC3 (DMIC01)
./tools/topology/sof-cnl-nocodec.m4:PCM_CAPTURE_ADD(DMIC01, 3, PIPELINE_PCM_7)
./tools/topology/sof-cnl-rt274.m4:VIRTUAL_WIDGET(DMIC01 Rx, out_drv, 9)
./tools/topology/sof-ehl-rt5660.m4:# PCM1 <---- volume <----- DMIC01 (dmic0 capture)
./tools/topology/sof-ehl-rt5660.m4:# capture DAI is DMIC01 using 2 periods
./tools/topology/sof-ehl-rt5660.m4:PCM_CAPTURE_ADD(DMIC01, 1, PIPELINE_PCM_3)
./tools/topology/sof-ehl-rt5660-nohdmi.m4:# PCM1 <---- volume <----- DMIC01 (dmic0 capture)
./tools/topology/sof-ehl-rt5660-nohdmi.m4:# capture DAI is DMIC01 using 2 periods
./tools/topology/sof-ehl-rt5660-nohdmi.m4:PCM_CAPTURE_ADD(DMIC01, 1, PIPELINE_PCM_3)
./tools/topology/sof-glk-da7219-kwd.m4:PCM_CAPTURE_ADD(DMIC01, 99, PIPELINE_PCM_4)
./tools/topology/sof-glk-da7219-kwd.m4:VIRTUAL_WIDGET(DMIC01 Rx, out_drv, 3)
./tools/topology/sof-glk-da7219.m4:PCM_CAPTURE_ADD(DMIC01, 99, PIPELINE_PCM_4)
./tools/topology/sof-glk-da7219.m4:VIRTUAL_WIDGET(DMIC01 Rx, out_drv, 3)
./tools/topology/sof-glk-rt5682.m4:PCM_CAPTURE_ADD(DMIC01, 99, PIPELINE_PCM_4)
./tools/topology/sof-glk-rt5682.m4:VIRTUAL_WIDGET(DMIC01 Rx, out_drv, 3)
./tools/topology/sof-icl-dmic-4ch.m4:# PCM6 <---- volume <----- DMIC01 (dmic0 capture)
./tools/topology/sof-icl-dmic-4ch.m4:PCM_CAPTURE_ADD(DMIC01, 1, PIPELINE_PCM_1)
./tools/topology/sof-icl-rt700.m4:# PCM2 <---------------- DMIC01 (dmic0 capture, BE dailink 2)
./tools/topology/sof-icl-rt700.m4:# capture DAI is DMIC01 using 2 periods
./tools/topology/sof-icl-rt700.m4:PCM_CAPTURE_ADD(DMIC01, 2, PIPELINE_PCM_3)
./tools/topology/sof-tgl-nocodec.m4:# PCM6 <---- volume <----- DMIC6 (DMIC01)
./tools/topology/sof-tgl-nocodec.m4:PCM_CAPTURE_ADD(DMIC01, 6, PIPELINE_PCM_13)
./tools/topology/sof-tgl-rt711-rt1308.m4:# PCM3 <---- volume <----- DMIC01 (dmic0 capture) BE dailink 3
./tools/topology/sof-tgl-rt711-rt1308.m4:# capture DAI is DMIC01 using 2 periods
./tools/topology/sof-tgl-rt711-rt1308.m4:PCM_CAPTURE_ADD(DMIC01, 3, PIPELINE_PCM_4)
```

Reply:
see this one :

````
platform/intel/intel-generic-dmic.m4:PCM_CAPTURE_ADD(DMIC48kHz, DMIC_DAI_LINK_48k_ID, concat(`PIPELINE_PCM_', DMIC_PIPELINE_48k_ID))
platform/intel/intel-generic-dmic.m4:PCM_CAPTURE_ADD(DMIC16kHz, DMIC_DAI_LINK_16k_ID, concat(`PIPELINE_PCM_', DMIC_PIPELINE_16k_ID))
````

We need to make sure the names are aligned for e.g. UCM to work. But this could be another PR.

Reply:
@plbossart Sure. I'm working on it now.

Reply:
@plbossart I saw  DAI_ADD, PCM_CAPTURE_ADD and DAI_CONFIG contains DMIC01/dmic01. Do you think we should replace them all or just replace DMIC01 with DMIC48k in PCM_CAPTURE_ADD. If only replace PCM_CAPTURE_ADD(), it seems we don't need to change the driver and UCM?

Reply:
I don't understand the question. There is no driver change required and no UCM file so far?
The PCM name should be DMIC48kHz, DMIC 01 means nothing to the user.

Reply:
If we only change the name in PCM_PLAYBACK_ADD(DMIC01) to PCM_PLAYBACK_ADD(DMIC48k), we don't need the change in driver. But if we also did the change in other place, such as DAI_ADD(dmic01) to DAI_ADD(dmic48k), we need to change the driver.